cmd.c

```diff
@@ -843,6 +843,75 @@ void cmd_syspage(char *s)
 }
 
 
+void cmd_phfs(char *s)
+{
+	u16 argsc = 0;
+	unsigned int i, major, minor, pos = 0;
+	char args[5][LINESZ + 1];
+
+	for (i = 0; argsc < 5; ++i) {
+		if (cmd_getnext(s, &pos, ". \t", NULL, args[i], sizeof(args[i])) == NULL || *args[i] == 0)
+			break;
+		argsc++;
+	}
+
+	if (argsc < 3 || argsc > 4) {
+		plostd_printf(ATTR_ERROR, "\nWrong arguments!!\n");
+		return;
+	}
```

Comment:
First we parse arguments, then we check how many there were? I think that `if` and `for` should switch places.

Reply:
As you can see in `for` loop there is incrementation of `argsc`. We don't  know how many arguments is in the command so firstly you have to parse them and after that check the count.

Reply:
Yeah, you're right, I missed that. There is no real need for `i` in this loop. How about:

```
for (argsc  = 0; argsc < 5; ++argsc ) {
	if (cmd_getnext(s, &pos, ". \t", NULL, args[argsc], sizeof(args[argsc])) == NULL || *args[argsc] == '\0')
		break;
}
```

Reply:
Done